README.md

```diff
@@ -1,65 +1,112 @@
-We are initially targeting Go and Rust.
-See [Tutorial](docs/tutorial.md) for some examples of starting `bpfd`, managing logs, and using `bpfctl`.
-
-In order to allow the attachment of multiple XDP programs to the same interface, we have implemented the
-[libxdp multiprog protocol](https://github.com/xdp-project/xdp-tools/blob/master/lib/libxdp/protocol.org).
-Offering this in bpfd allows for XDP applications whose loader was not using libxdp to benefit from this.
-We are also hoping to find a way for applications linked with libxdp to use bpfd instead if it's
-in use in the system.
-
-## Usage
-
-Load a sample XDP Program:
-
-```console
-cargo build
-sudo ./target/debug/bpfd&
-sudo ./target/debug/bpfctl load --path /path/to/xdp/program --section-name "pass" xdp -i wlp2s0 --priority 50
-```
+### Why BPF?
+
+BPF is a powerful general-purpose framework that allows running sandboxed
+programs in the kernel.  It can be used for many purposes, including networking,
+monitoring, tracing and security.
+
+### Why BPF in Kubernetes?
+
+Demand is increasing from both Kubernetes developers and users. Examples of bpf
+in kubernetes include:
+
+- [Cilium](https://cilium.io/) and
+  [Calico](https://www.tigera.io/project-calico/) CNIs 
+- [Pixe](https://px.dev/): Open source observability
+- [KubeArmor](https://kubearmor.io/): Container-aware Runtime Security
+  Enforcement System
+- [Blixt](https://github.com/Kong/blixt): Gateway Api L4 conformance
+  implementation
+- [NetObserv](https://github.com/netobserv): Open Source Operator for network
+  observability 
+
+### Challenges for BPF in Kubernetes
+
+- Requires privileged pods
+  - BPF-enabled apps require at least CAP_BPF permissions, and more (depending
+    on what they need to do).
+  - These permissions give the pods great powers.
+- Handling multiple BPF programs
+  - Not all libraries support multiple XDP programs sharing the same interface.
+  - When multiple BPF programs are used, they can interfere with each other in
+    unpredictable ways.
+- Debugging BPF-related problems is difficult -- particularly on Kubernetes.
+- There are no standard libraries for loading and managing BPF programs leaving
+  each application developer to implement their own.
+- BPF byte code often needs to be compiled into associated applications.
+
+### What is bpfd?
+
+bpfd is a system daemon that supports, loading, unloading, modifying, and
+monitoring BPF programs. The solution for Kubernetes uses an operator, built
+with the Operator SDK, that can deploy the bpfd daemon and a bpfd agent to all
+the nodes in a cluster. The agent supports well-defined Kubernetes APIs (CRDs)
+that can be used to manage the lifecycle of BPF programs across a Kubernetes
+cluster.
+
+bpfd is developed in Rust and built on top of Aya, a Rust BPF library.
+
+The benefits of this solution include the following:
+
+- Security
+  - Improved security because only the bpfd daemon, which can be tightly
```

Comment:
You could mention here that only a single thread keeps these capabilities and the other threads (serving RPCs) do not.